### PR TITLE
Fix refunds on Stripe API 2022-11-15

### LINKED ISSUE #2497: Refund incompatibility with webhooks using Stripe API 2022-11-15
**Describe the bug**

Orders will fail to refund for incoming refund webhooks on the latest Stripe API, most probably due to [this](https://stripe.com/docs/upgrades#2022-11-15):

> Charge no longer auto-expands refunds by default. You can expand the list but for performance reasons we recommended against doing so unless needed.

Right now, the following error is thrown:

```
Undefined property: stdClass::$refunds in includes/class-wc-stripe-webhook-handler.php
```

**To Reproduce**
1. Add a new webhook endpoint to the Stripe dashboard using version 2022-11-15.
2. As a shopper, checkout with a regular test credit card.
3. From WP-admin, click the recently created order, and then open its charge ID on Stripe.
4. Refund the order from the Stripe dashboard.
5. Notice the order status is not changed from WC, and the above error appears in WC status logs.

**Expected behavior**

Refunds should be compatible with the latest Stripe API version.

### PR DESCRIPTION
**Pull request by ricardo**
Fixes #2497 
Related to #2493

TL;DR: Starting from [API version 2022-11-15](https://stripe.com/docs/upgrades#2022-11-15), the Charge object no longer auto-expands refunds by default.

In the plugin, the API version is [overwritten](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/1833465bc14d3a699abec9ec7d4f64b0fbdb4f24/includes/class-wc-stripe-api.php#L17) when making requests to the Stripe API, but for incoming webhooks, there's no way to overwrite the version, causing the webhooks to fail. This PR makes the webhook handler compatible with refunds.

## Changes proposed in this Pull Request:

- Explicitly expand `refunds` when retrieving a Charge object from a webhook notification.

## Testing instructions

You'll need a standard Stripe account with its [API version](https://dashboard.stripe.com/developers) set to anything prior to [2022-11-15](https://stripe.com/docs/upgrades#2022-11-15) (not marked as default in the dashboard). The WCPay Dev (Express) account can be used if you don't have a personal test account.

1. Expose your local environment publicly. You can use Ngrok or JT. (For valet: `cd valet/stripe && valet unsecure && valet share`).
2. Create a new webhook endpoint from the [Stripe dashboard](https://dashboard.stripe.com/test/webhooks).
   i. Use the URL from the **Stripe plugin settings > Settings > Account details**.
   ii. **Add events > Select all events**.
   iii. Select API version `2022-11-15`.
3. Add the endpoint and copy the webhook signing secret. Paste the secret into the plugin test webhook secret.
4. As a shopper, checkout with card.
5. Ensure refunds work:
   a. Ensure refunding from the Stripe dashboard works (Click the charge ID from the WC order and click "Refund" from the Stripe dashboard).
   b. Ensure refunding from "Edit order" works (**WC > Orders > Edit order > Refund > Refund `X` via Stripe**).
   c. Ensure partial refunds work from the Stripe dashboard and from Edit order. (Once the order is completely refunded, its status is changed to "Refunded").
   d. Ensure refunding a charge authorization works. ("Issue an authorization on checkout, and capture later" must be enabled from WC Stripe settings).

**Note:** To make sure a refund works:
- The charge status must be set to "refunded" in the Stripe dashboard;
- No errors should be thrown in Stripe logs;
- The status must be changed to "Refunded" when the order is completely refunded, and;
- There must be no errors in the order notes.

To make sure the changes are BW compatible with API versions prior to `2022-11-15`, please disable the `2022-11-15` webhook endpoint, create a new one using a previous API version (don't forget to change the webhook signing secret), and test steps 4 and 5.

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: includes/class-wc-stripe-webhook-handler.php [hunk 1]
```diff
@@ -768,6 +770,25 @@ public function is_partial_capture( $notification ) {
...
+	 * @param object $notification
+	 *
+	 * @return object
+	 */
+	public function get_refund_object( $notification ) {
+		// Since API version 2022-11-15, the Charge object no longer expands `refunds` by default.
+		// We can remove this once we drop support for API versions prior to 2022-11-15.
+		if ( ! empty( $notification->data->object->refunds->data[0] ) ) {
+			return $notification->data->object->refunds->data[0];
+		}
```
Comment (ricardo) [inline includes/class-wc-stripe-webhook-handler.php:786]: This isn't required as the snippet below should expand the refunds list from the Charge object, but it's one less request and we can keep it until we drop support for API versions prior to 2022-11-15.

File: includes/class-wc-stripe-webhook-handler.php [hunk 2]
```diff
@@ -768,6 +770,25 @@ public function is_partial_capture( $notification ) {
...
+
+		$charge = $this->get_charge_object( $notification->data->object->id, [ 'expand' => [ 'refunds' ] ] );
```
Comment (ricardo) [inline includes/class-wc-stripe-webhook-handler.php:788]: The `'expand' => [ 'refunds' ]` parameter here isn't required while we're still using version `2019-09-09` from our requests to Stripe [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/trunk/includes/class-wc-stripe-api.php#L17).

The current version will automatically expand refunds, but once we move to `2022-11-15` or later, this is going to be ultimately required, so to prevent future issues, I think it's better to be explicit about expanding the refunds property now.

**Review by rafaelzaleski — APPROVED** [review]
Looks good and tests well! 👍
